### DISCUSSION TITLE
Backup Methods Supported?

### DISCUSSION THREAD

**Discussion by excessi0n** [eduardogsilva/routerfleet#5] [Q&A] [answered]
Hi, reading through your readme I couldn't find any mention of what backup method(s) you've implemented to do the actual config retrieval? Do you think you could add a section for that? Are any vendor agnostic methods supported?

Is there any other documentation I've missed?

Many thanks.

**Comment by eduardogsilva** [comment]
Hi @excessi0n at this moment, my documentation is a bit limited.

regarding the backups, I’ve created a generic model that is able to store a text backup and a file based backup.

At this moment, routerfleet just supports mikrotik devices, but when I start adding new compatible devices, I will study case by case.

currently, I’m retrieving both formats from mikrotik router. The configuration export and the binary backup.

the idea is to keep this standard for next devices.

**Reply by excessi0n** [reply]
Hi, thanks for the quick response. 

I'm actually curious how you're initiating and then retrieving the backups? 
To be clear; here's a list of ways I can think of you might be doing it:

SSH / SCP, API, REST API, FTP / SFTP, TFTP, Scripted on the device and pushed to your repository, POST would also be an option this way.

I'm trying to gauge what level of port configuration / additional attack surface exposure I'd need to account for.

Thanks :)

**Comment by eduardogsilva** [comment, marked as answer]
@excessi0n the backups are initiated over SSH and are retrieved using scp (same ssh port). Actually, you shouldn't expose any ports of your mikrotik to the internet. Just keep everything locked down on your management network if you have one. 

If you want to backup remote mikrotiks and keep everything locked down, take a look on my other project. 
I've built wireguard_webadmin mostly for connecting my mikrotiks. There's also a nice firewall included ;)

**Reply by excessi0n** [reply]
Great, thank you.
